tests/schema/tests.py

```diff
@@ -1355,3 +1355,47 @@ def test_custom_manytomanyfield(self):
         finally:
             # Cleanup model states
             AuthorWithM2M._meta.local_many_to_many.remove(new_field)
+
+    def test_inherited_manytomanyfield(self):
+        """
+        #24104 - Schema editors should look for internal type of field
+        """
+        # Create the tables
+        with connection.schema_editor() as editor:
+            editor.create_model(AuthorWithM2M)
+            editor.create_model(TagM2MTest)
+        # Create an M2M field
+        new_field = InheritedManyToManyField("schema.TagM2MTest", related_name="authors")
+        new_field.contribute_to_class(AuthorWithM2M, "tags")
+        # Ensure there's no m2m table there
+        self.assertRaises(DatabaseError, self.column_classes, new_field.rel.through)
+        try:
+            # Add the field
+            with connection.schema_editor() as editor:
+                editor.add_field(
```

Comment:
not sure if this test needs to be forward ported, but I think we agreed to single line these in a subsequent cleanup PR. Otherwise good to go.